File: Determine_if_two_trees_are_identical.java
class Solution{
	boolean isIdentical(Node root1, Node root2){
	    if(root1 == null && root2 == null){
	        return true;
	    }
	    if(root1 == null || root2 == null){
	        return false;
	    }
	    return root1.data == root2.data && isIdentical(root1.left, root2.left) && 
	    isIdentical(root1.right, root2.right);
	}
}

File: Determine_if_two_trees_are_identical.cpp
class Solution{
    public:    
    bool isIdentical(Node *r1, Node *r2){
        if(!r1 and !r2){
            return true;
        }
        if(!r1 or !r2){
            return false;
        }
        return r1 -> data == r2-> data and 
        isIdentical(r1 -> left, r2 -> left) and isIdentical(r1 -> right, r2 -> right);
    }
};
